e2e-tests/ukef/cypress/e2e/journeys/portal/deal-cancellation/gef/deal-cancellation-maker-unable-issue-facility.spec.js

```diff
@@ -0,0 +1,153 @@
+import { DEAL_STATUS, DEAL_SUBMISSION_TYPE, TFM_URL } from '@ukef/dtfs2-common';
+import relative from '../../../../relativeURL';
+import gefPages from '../../../../../../../gef/cypress/e2e/pages';
+import MOCK_USERS from '../../../../../../../e2e-fixtures/portal-users.fixture';
+import { MOCK_APPLICATION_AIN_DRAFT, MOCK_APPLICATION_MIN_DRAFT } from '../../../../../../../e2e-fixtures/gef/mocks/mock-deals';
+import { anIssuedCashFacility, anUnissuedCashFacility, multipleMockGefFacilities } from '../../../../../../../e2e-fixtures/mock-gef-facilities';
+import { yesterday, tomorrow } from '../../../../../../../e2e-fixtures/dateConstants';
+
+import { PIM_USER_1 } from '../../../../../../../e2e-fixtures';
+
+const { BANK1_MAKER1 } = MOCK_USERS;
+
+context('When a deal has been cancelled on TFM, maker unable to issue facility', () => {
+  const ainDeals = Array(4).fill(MOCK_APPLICATION_AIN_DRAFT);
+  const minDeals = Array(4).fill(MOCK_APPLICATION_MIN_DRAFT);
+  const gefDeals = [...ainDeals, ...minDeals];
+  const deals = [];
+
+  const facilities = [
+    anIssuedCashFacility({ facilityEndDateEnabled: true }),
+    anIssuedCashFacility({ facilityEndDateEnabled: true }),
+    anUnissuedCashFacility({ facilityEndDateEnabled: true }),
+    multipleMockGefFacilities({ facilityEndDateEnabled: true }).unissuedContingentFacility,
+    anIssuedCashFacility({ facilityEndDateEnabled: true }),
+    anIssuedCashFacility({ facilityEndDateEnabled: true }),
+    multipleMockGefFacilities({ facilityEndDateEnabled: true }).unissuedContingentFacility,
+    anUnissuedCashFacility({ facilityEndDateEnabled: true }),
+  ];
+
+  before(() => {
+    cy.insertManyGefDeals(gefDeals, BANK1_MAKER1).then((insertedDeals) => {
```

Comment:
This is too complicated for before, can this be a support command please.

Reply:
@abhi-markan I suggest we leave as it is - as much as I'm not a fan, it's the current approach for all GEF E2E tests.

Other tests are more complex, e.g `review-ukef-approval-with-issued-and-unissued-facilities-MIA-and-resubmit-to-ukef.spec`

Ultimately, we shouldn't be calling the API and inserting deals/facilities - it should be going through the UI instead - there is a ticket for this DTFS2-7435.

Whether this is in a command or a `before` - we still have the same fundamental issue.

I feel that adding a command here for this test will misalign things.